include/kernels/dslash_domain_wall_5d.cuh

```diff
@@ -10,16 +10,22 @@ namespace quda
   template <typename Float, int nColor, int nDim, QudaReconstructType reconstruct_>
   struct DomainWall5DArg : WilsonArg<Float, nColor, nDim, reconstruct_> {
     typedef typename mapper<Float>::type real;
-    int Ls;   /** fifth dimension length */
+    int_fastdiv Ls;   /** fifth dimension length */
     real a;   /** xpay scale factor */
     real m_f; /** fermion mass parameter */
 
-    DomainWall5DArg(ColorSpinorField &out, const ColorSpinorField &in, const GaugeField &U, double a, double m_f,
-                    bool xpay, const ColorSpinorField &x, int parity, bool dagger, const int *comm_override) :
-      WilsonArg<Float, nColor, nDim, reconstruct_>(out, in, U, xpay ? a : 0.0, x, parity, dagger, comm_override),
+    DomainWall5DArg(cvector_ref<ColorSpinorField> &out, cvector_ref<const ColorSpinorField> &in, const ColorSpinorField &halo,
+                    const GaugeField &U, double a, double m_f, bool xpay, cvector_ref<const ColorSpinorField> &x, int parity,
+                    bool dagger, const int *comm_override) :
+      WilsonArg<Float, nColor, nDim, reconstruct_>(out, in, halo, U, xpay ? a : 0.0, x, parity, dagger, comm_override),
       Ls(in.X(4)),
       a(a),
-      m_f(m_f) {}
+      m_f(m_f)
+    {
+      // FIXME this is a hack to use remove the batch dimension from the dslash constants
```

Comment:
A reminder to fix this :)

Reply:
I've clarified the comment there: it's not so much of a hack, and is the right way to do this.

Looking at this made me realize there was a bug with the flops counter for the 5-d DWF kernel, so I've fixed that (23c6ba0e668c1b4ab205eb72ac4f6fa752223c6a)